### DISCUSSION TITLE
refetchOnWindowFocus: always not triggering, any ideas?

### DISCUSSION THREAD

**Discussion by macutko** [TanStack/query#6546] [Q&A] [answered]
Hi, I am fairly new to this library. I have a query with the refetchOnWindowFocus set to always. I can see this reported in the devtools as well. However, when I click out of the chrome window and back in, I do not see a request fired in my Network tab. The query is marked as stale and the refetchInterval does trigger a refetch after 6000ms. So I am not really sure why this is happening. Any ideas on how to start debugging this? I tried putting a function instead of "always" but that never seems to get triggered.

**Comment by TkDodo** [comment, marked as answer]
if you are using v5, window focus is only listening to the `visibilitychange` event, so you have to really switch tabs to see it getting triggered.

this is listed in the [migration guide](https://tanstack.com/query/latest/docs/react/guides/migrating-to-v5#window-focus-refetching-no-longer-listens-to-the-focus-event) and it fixed a couple of bugs as outlined [here](https://github.com/TanStack/query/pull/4805).

You can still set a [custom event listener](https://tanstack.com/query/latest/docs/react/reference/focusManager#focusmanagerseteventlistener) and listen to the `focus` event if you want.
